synapse/models/planning.py

```diff
@@ -0,0 +1,155 @@
+                        'doc': 'The time the planning system was last updated.'}),
+
+                    ('version', ('it:semver', {}), {
+                        'doc': 'The version of the planning system.'}),
+
+                    ('url', ('inet:url', {}), {
+                        'doc': 'The primary URL which documents the planning system.'}),
+                )),
+                ('plan:phase', {}, (
+                    ('title', ('str', {}), {
+                        'ex': 'Reconisence Phase',
+                        'doc': 'The title of the phase.'}),
+
+                    ('summary', ('str', {}), {
+                        'doc': 'A summary of the definition of the phase.'}),
+
+                    ('index', ('int', {}), {
+                        'doc': 'The index of this phase within the phases of the system.'}),
+
+                    ('url', ('inet:url', {}), {
+                        'doc': 'A URL which links to the full documentation about the phase.'}),
+
+                    ('system', ('plan:system', {}), {
+                        'doc': 'The planning system which defines this phase.'}),
+                )),
+                ('plan:procedure:type:taxonomy', {}, ()),
+                ('plan:procedure', {}, (
+
+                    ('title', ('str', {}), {
+                        'ex': 'Network Reconnaissance Procedure',
+                        'doc': 'The name of the procedure.'}),
+
+                    ('summary', ('str', {}), {
+                        'doc': 'A summary of the purpose and use cases for the procedure.'}),
+
+                    ('author', ('ps:contact', {}), {
+                        'doc': 'The contact of the person or organization which authored the procedure.'}),
+
+                    ('created', ('time', {}), {
+                        'doc': 'The time the procedure was created.'}),
+
+                    ('updated', ('time', {}), {
+                        'doc': 'The time the procedure was last updated.'}),
+
+                    ('version', ('it:semver', {}), {
+                        'doc': 'The version of the procedure.'}),
+
+                    ('system', ('plan:system', {}), {
+                        'doc': 'The planning system which defines this procedure.'}),
+
+                    ('type', ('plan:procedure:type:taxonomy', {}), {
+                        'doc': 'A type classification for the procedure.'}),
+
+                    ('inputs', ('array', {'type': 'plan:procedure:variable', 'uniq': True, 'sorted': True}), {
+                        'doc': 'An array of inputs required to execute the procedure.'}),
+
+                    ('firststep', ('plan:procedure:step', {}), {
+                        'doc': 'The first step in the procedure.'}),
+                )),
+                ('plan:procedure:variable', {}, (
+
+                    ('name', ('str', {}), {
+                        'doc': 'The name of the variable.'}),
+
+                    ('type', ('str', {}), {
```

Comment:
Worth making this a taxonomy type?

Reply:
Every time we do type as a `str`, we almost always regret it.

Reply:
This is not a classic use of `:type` like taxonomy, this is designed to capture a type from a programatic type system such as `int` and `str` so a taxonomy does not make sense here. 👍

Reply:
Do we want to `onespace` and `lower` the value then? Avoid `Int` vs `int` vs ` int ` problems.